winch/codegen/src/isa/x64/masm.rs

```diff
@@ -1118,12 +1151,21 @@ impl MacroAssembler {
         shared_flags: settings::Flags,
         isa_flags: x64_settings::Flags,
     ) -> Self {
+        let ptr_type: WasmValType = ptr_type_from_ptr_size(ptr_size.size()).into();
+        // Flags used for WebAssembly loads / stores.
+        // Untrusted by default so we don't set `no_trap`.
+        // We also ensure that the endianess is the right one for WebAssembly.
+        let mut untrusted_flags = MemFlags::new();
+        untrusted_flags.set_endianness(Endianness::Little);
+
         Self {
             sp_offset: 0,
             asm: Assembler::new(shared_flags.clone(), isa_flags.clone()),
             flags: isa_flags,
             shared_flags,
-            ptr_size: ptr_type_from_ptr_size(ptr_size.size()).into(),
+            ptr_size: ptr_type.into(),
+            trusted_flags: MemFlags::trusted(),
```

Comment:
Yeah, this could definitely be improved I guess, the current implementation is already an improvement to what was there before in the sense that we were creating a `MemFlags::trusted` per load / store, now it's created once per MacroAssembler (and by definition once per function). Given that the flags won't be modified, I think that sharing both (trusted and untrusted) through all the functions might make sense. That said, it was not evident to me how to go about the constant route, were you thinking about introducing something like `OnceCell`? Having a pure constant won't work given that `MemFlags::trusted` is not `const`. Another approach could be to create a `MacroAssemblerContext` that is created verly early in the process and passed in to each `MacroAssembler` implementation and reused across all the functions.